MapleServer2/Types/ItemStats.cs

```diff
@@ -109,28 +109,28 @@ public ItemStats(int itemId, int grade)
                 }
             }
 
-            if (ItemStatsMetadataStorage.GetRandomStat(itemId, out List<ItemOptions> randomList))
+            if (ItemStatsMetadataStorage.GetRandomBonus(itemId, out List<ItemOptions> randomBonusList))
             {
-                foreach (ItemOptions item in randomList)
+                foreach (ItemOptions options in randomBonusList)
```

Comment:
Instead of iterating over all the list you could use:

```suggestion
randomBonusList.Find(options => options.Rarity == rarity && options.Slots > 0));
```

We dont want multiple options, just the one with the same rarity and at least 1 slot.
Return immediately if no options is found.

Reply:
Same principle apply to `basicList`